### PR TITLE
fix(opensearch): cannot disable cluster logging

### LINKED ISSUE #29200: aws-opensearchservice: cannot disable logging
### Describe the bug

Enabling logging on OpenSearch clusters will populate LogPublishingOptions in the output template, but disabling logging leaves LogPublishingOptions empty so the cluster will retain the previous logging config. I don't see any documentation saying that enabling logging is a one-way trip, so I think this is a bug.

Specifically, I'm talking about these logging props:
```
new es.Domain(scope, id, {
 ...
 logging: {
    slowSearchLogEnabled: false,
    appLogEnabled: false,
    slowIndexLogEnabled: false,
  },
})
```

### Expected Behavior

```
    "LogPublishingOptions": {
        "ES_APPLICATION_LOGS": {
            "Enabled": false
        },
        "SEARCH_SLOW_LOGS": {
            "Enabled": false
        },
        "INDEX_SLOW_LOGS": {
            "Enabled": false
        }
    },
```

### Current Behavior

```
    "LogPublishingOptions": {},
```

### Reproduction Steps

```
const { Stack, Duration } = require('aws-cdk-lib');
const es = require('aws-cdk-lib/aws-opensearchservice');

class PocStack extends Stack {
  constructor(scope, id, props) {
    super(scope, id, props);

    const cluster = new es.Domain(this, 'OpenSearchCluster', {
      domainName: 'test-opensearch-cluster',
      version: es.EngineVersion.OPENSEARCH_2_11,
      capacity: {
        dataNodes: 1,
        dataNodeInstanceType: "t4g.medium.search",
      },
      ebs: {
        volumeSize: 10,
      },
      logging: {
        appLogEnabled: true,
      }
    });
  }
}

module.exports = { PocStack }
```

### Possible Solution

If you don't need the current behaviour, then have it perform the expected behaviour from above.

If there is a good reason it works that way or you want backwards-compatibility, then maybe have logging props like appLogEnabled default to some non-boolean value (e.g. `-1`). If left default (not specified by user), then LogPublishingOptions can be empty. If the user explicitly specifies either true or false, then populate the LogPublishingOptions to enable or disable logging as appropriate.

### Additional Information/Context

An environment hit the limit on cloudwatch resource policies, so I'm working on a change related to the new (much appreciated) `suppressLogsResourcePolicy` prop. The way things are currently, I'll have to manually (*shudder*) disable logging on a cluster to clear up policy space before I can deploy the new multi-cluster logging resource policy.

### CDK CLI Version

2.128.0

### Framework Version

_No response_

### Node.js Version

v18.18.2

### OS

OSX

### Language

TypeScript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by sigJoe**
### Issue # (if applicable)

#29294

### Reason for this change

Currently cannot disable opensearch logging

### Description of changes

If log parameters are explicitly set to false rather than undefined, it populates the logPublishingOptions with config to disable that logging

### Description of how you validated changes

I added unit tests, although to be honest jest is giving me lots of trouble and I'm out of time for the day so I'll just create this pR and see what happens. Fingers crossed the PR test check is clean and I can pretend I know what I'm doing.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by SigniantJoe** [comment]
Clarification Request: Please advise. I did my best before asking, but I don't understand what the PR checks are complaining about or how to fix it.

File: tests.txt
Comment (lpizzinidev) [inline tests.txt:1]: Why is this file included? 🤔

**Review by lpizzinidev — APPROVED** [review]
Looks good 👍
There is only one file that should be removed from the PR imo.
Also, can you please update the title to [describe the bug (not the solution)](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md#step-4-pull-request).

File: tests.txt
Comment (SigniantJoe) [reply tests.txt:1]: I based my test on what I thought were existing patterns, and this file contained a list of all the tests so I inserted mine as well
/shrug

I'll remove it now

File: tests.txt [hunk 1]
```diff
@@ -1 +1 @@
-@aws-cdk-testing/framework-integ/test/aws-s3/test/integ.bucket-auto-delete-objects.js
+@aws-cdk-testing/framework-integ/test/aws-s3/test/integ.bucket-auto-delete-objects.js
```
Comment (GavinZZ) [inline tests.txt:1]: curious why GH is highlighting this change but they seem to be identical?

**Review by GavinZZ — COMMENTED** [review]
Changes look good, one question in mind before approving.

File: tests.txt (hunk 1 above)
Comment (SigniantJoe) [reply tests.txt]: It was missing a newline. Sorry about that. It's fixed now

**Review by GavinZZ — APPROVED** [review]

**Review by GavinZZ — APPROVED** [review]
